corehq/apps/smsbillables/management/commands/bootstrap_twilio_gateway.py

```diff
@@ -7,101 +7,124 @@
+                network = mach_table.cell_value(row, 5).lower().replace(' ', '')
+                subscribers = 0
+                try:
+                    subscribers = int(mach_table.cell_value(row, 10).replace('.', ''))
+                except ValueError:
+                    logger.info("Incomplete subscriber data for country code %d" % country_code)
+                if not(iso in mach_data):
+                    mach_data[iso] = {}
+                mach_data[iso][network] = (country_code, subscribers)
+                row += 1
+        except IndexError:
+            pass
+        return mach_data
+
+    twilio_data = get_twilio_data()
+    mach_data = get_mach_data()
+
+    for iso in twilio_data:
+        if iso in mach_data:
+            weighted_price = 0
+            total_subscriptions = 0
+            country_code = None
+            calculate_other = False
+            for twilio_provider in twilio_data[iso]:
+                if twilio_provider == 'other':
+                    calculate_other = True
                 else:
-                    try:
-                        iso = row[0].lower()
-                        provider = row[1].split('-')[1].lower().replace(' ', '')
-                        rate = float(row[2])
-                        if not(iso in twilio_data):
-                            twilio_data[iso] = {}
-                        twilio_data[iso][provider] = rate
-                    except IndexError:
-                        logger.info("Twilio index error %s:" % row)
-            twilio_file.close()
-            return twilio_data
+                    for mach_provider in mach_data[iso]:
+                        try:
+                            if twilio_provider in mach_provider:
+                                country_code, subscriptions = mach_data[iso][mach_provider]
+                                weighted_price += twilio_data[iso][twilio_provider] * subscriptions
+                                total_subscriptions += subscriptions
+                                mach_data[iso][mach_provider] = country_code, 0
+                                break
+                        except UnicodeDecodeError:
+                            pass
+            if calculate_other:
+                other_rate_twilio = twilio_data[iso]['other']
+                for _, subscriptions in mach_data[iso].values():
+                    weighted_price += other_rate_twilio * subscriptions
+            if country_code is not None:
+                weighted_price = weighted_price / total_subscriptions
+                SmsGatewayFee.create_new(
+                    TwilioBackend.get_api_id(),
+                    OUTGOING,
+                    weighted_price,
+                    country_code=country_code,
+                    currency=currency_class.objects.get(code="USD"),
+                    fee_class=sms_gateway_fee_class,
+                    criteria_class=sms_gateway_fee_criteria_class,
+                )
```

Comment:
that's correct - I'd recommend adding them in a new migration, since we don't want to re-run migrations on every running instance of hq